lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -682,3 +684,42 @@ bool swift::ArraySemanticsCall::replaceByValue(SILValue V) {
   removeCall();
   return true;
 }
+
+bool swift::ArraySemanticsCall::replaceByAppendingValues(SILModule &M,
+  SILFunction *AppendFn,
+  SmallVectorImpl<SILValue> &Vs,
+  ArrayRef<Substitution> Subs) {
+  assert(getKind() == ArrayCallKind::kAppendContentsOf &&
+         "Must be an append_contentsOf call");
+  assert(AppendFn && "Must provide an append SILFunction");
+
+  // We only handle loadable types.
+  for(auto &V : Vs) {
+    if (!V->getType().isLoadable(M))
+      return false;
+  }
+
+  auto ArrRef = SemanticsCall->getArgument(1);
+  SILBuilderWithScope Builder(SemanticsCall);
+  auto Loc = SemanticsCall->getLoc();
+  auto *FnRef = Builder.createFunctionRef(Loc, AppendFn);
+  auto FnTy = FnRef->getType();
+
+  for(auto& V : Vs) {
```

Comment:
'auto&' => 'auto &'

This is not LLVM style, please use git-clang-format, it will fix these issues for you in a consistent way.